### PR TITLE
Refactor based on cdap-e2e-tests framework and increase converage

### PR DESCRIPTION
**Pull request by poojantcs**

### CODE REVIEW

File: src/e2e-test/features/salesforcemultiobjectsbatchsource/RunTimeWithMacros.feature [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    And Select plugin: "Salesforce Multi Objects" from the plugins list as: "Source"
+    And Navigate to the properties page of plugin: "SalesforceMultiObjects"
+    And fill Reference Name property
+    And Click on the Macro button of Property: "username" and set the value to: "Username"
+    And Click on the Macro button of Property: "password" and set the value to: "Password"
+    And Click on the Macro button of Property: "securityToken" and set the value to: "SecurityToken"
+    And Click on the Macro button of Property: "consumerKey" and set the value to: "ConsumerKey"
+    And Click on the Macro button of Property: "consumerSecret" and set the value to: "ConsumerSecret"
+    And Click on the Macro button of Property: "loginUrl" and set the value to: "LoginUrl"
+    And Click on the Macro button of Property: "whiteList" and set the value to: "WhiteList1"
```
Comment (rmstar) [inline src/e2e-test/features/salesforcemultiobjectsbatchsource/RunTimeWithMacros.feature]: why are we setting whiteList param here? Aren't we testing black list set as macro param?

File: src/e2e-test/resources/pluginParameters.properties [hunk 2]
```diff
@@ -12,10 +12,33 @@ invalid.admin.password=def
...
+#SObject Query
+last.modified.after=2022-03-08T00:00:00Z
+last.modified.before=2022-03-21T00:00:00Z
+
+#Sink Properties
+sink.projectid=cdf-entcon
+externalId=1211
+
+TenthousandRecords=10000
+OnelackRecords=100000
```
Comment (rmstar) [inline src/e2e-test/resources/pluginParameters.properties]: nit: prefer HundredThousand, TenMillion, HundredMillion

File: src/main/java/io/cdap/plugin/salesforce/parser/SalesforceQueryParser.java [hunk 3]
```diff
@@ -16,6 +16,8 @@
 package io.cdap.plugin.salesforce.parser;
 
 import io.cdap.plugin.salesforce.SObjectDescriptor;
+
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/salesforce/parser/SalesforceQueryParser.java]: why are we changing src/main/java? This should be a test only change

File: src/e2e-test/java/io/cdap/plugin/utils/enums/ErrorHandlingOptions.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+package io.cdap.plugin.utils.enums;
+
+/**
+ * ErrorHandlingOptions enum.
+ */
+public enum ErrorHandlingOptions {
+  SKIP_ON_ERROR("Skip on error"),
+  STOP_ON_ERROR("Stop on error");
```
Comment (rmstar) [inline src/e2e-test/java/io/cdap/plugin/utils/enums/ErrorHandlingOptions.java:24]: are we testing SKIP_ON_ERROR vs STOP_ON_ERROR behavior? Also, I couldn't find STOP_ON_ERROR being used anywhere

File: src/e2e-test/features/salesforcestreamingsource/RunTimeWithMacros.feature [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    And Enter runtime argument value "admin.security.token" for key "SecurityToken"
+    And Enter runtime argument value "admin.consumer.key" for key "ConsumerKey"
+    And Enter runtime argument value "admin.consumer.secret" for key "ConsumerSecret"
+    And Enter runtime argument value "login.url" for key "LoginUrl"
+    And Enter runtime argument value "salesforce.streaming.TopicName" for key "TopicName"
+    And Enter runtime argument value "Salesforce.sobjectName" for key "SObjectName"
+    And Run the Pipeline in Runtime with runtime arguments
+    And Wait till pipeline is in running state
+    And Open and capture logs
+    And Verify the pipeline status is "Succeeded"
```
Comment (rmstar) [inline src/e2e-test/features/salesforcestreamingsource/RunTimeWithMacros.feature]: How does streaming pipeline go to succeeded state? Won't it remain in Running state until you stop it?

File: src/e2e-test/features/salesforcemultiobjectsbatchsource/RunTimeWithMacros.feature (hunk 1 above)
Comment (poojantcs) [reply src/e2e-test/features/salesforcemultiobjectsbatchsource/RunTimeWithMacros.feature]: Updated.

File: src/e2e-test/resources/pluginParameters.properties (hunk 2 above)
Comment (poojantcs) [reply src/e2e-test/resources/pluginParameters.properties]: Updated.

File: src/main/java/io/cdap/plugin/salesforce/parser/SalesforceQueryParser.java (hunk 3 above)
Comment (poojantcs) [reply src/main/java/io/cdap/plugin/salesforce/parser/SalesforceQueryParser.java]: Might have been done by mistake. Reverted it.

File: src/e2e-test/java/io/cdap/plugin/utils/enums/ErrorHandlingOptions.java (hunk 4 above)
Comment (poojantcs) [reply src/e2e-test/java/io/cdap/plugin/utils/enums/ErrorHandlingOptions.java:24]: Added tests for the same.

File: src/e2e-test/features/salesforcestreamingsource/RunTimeWithMacros.feature (hunk 5 above)
Comment (poojantcs) [reply src/e2e-test/features/salesforcestreamingsource/RunTimeWithMacros.feature]: Updated.

File: src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+    try {
+      HttpPost httpPost = new HttpPost(TOKEN_URL);
+      httpPost.setEntity(new UrlEncodedFormEntity(loginParams));
+      HttpResponse response = httpClient.execute(httpPost);
+      final int statusCode = response.getStatusLine().getStatusCode();
+      String responseDetails = EntityUtils.toString(response.getEntity());
+      JSONObject jsonObjectOfResponse = (JSONObject) new JSONTokener(responseDetails).nextValue();
+      loginAccessToken = jsonObjectOfResponse.getString("access_token");
+      loginInstanceUrl = jsonObjectOfResponse.getString("instance_url");
```
Comment (rmstar) [inline src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java:77]: Isn't this same as TOKEN_URL?

File: src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+      logger.info("Error in establishing connection to Salesforce: " + ioException);
+    } catch (JSONException jsonException) {
+      logger.info("Error in processing the Response in JSON: " + jsonException);
+    }
+
+    return loginAccessToken;
+  }
+
+  public static String createLead(JSONObject leadObject) throws UnsupportedEncodingException {
+    getAccessToken();
```
Comment (rmstar) [inline src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java:91]: Use the token returned by getAccessToken().

File: src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+  private static final String GRANTTYPE = PluginPropertyUtils.pluginProp("grant.type");
+  private static final String CLIENTID = PluginPropertyUtils.pluginProp("admin.consumer.key");
+  private static final String CLIENTSECRET = PluginPropertyUtils.pluginProp("admin.consumer.secret");
+  private static final String REST_ENDPOINT = PluginPropertyUtils.pluginProp("rest.api.endpoint");
+  private static final String API_VERSION = PluginPropertyUtils.pluginProp("rest.api.version");
+  private static final Header prettyPrintHeader = new BasicHeader("X-PrettyPrint", "1");
+  private static String loginAccessToken = null;
+  private static String loginInstanceUrl = null;
+
+  public static String getAccessToken() {
```
Comment (rmstar) [inline src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java:60]: Make it private

File: src/e2e-test/java/io/cdap/plugin/utils/enums/ErrorHandlingOptions.java (hunk 4 above)
Comment (rmstar) [reply src/e2e-test/java/io/cdap/plugin/utils/enums/ErrorHandlingOptions.java:24]: Are we testing error scenarios with SKIP_ON_ERROR vs STOP_ON_ERROR?
i.e. error records are skipped if SKIP_ON_ERROR is set, but pipeline run fails if there are errors and STOP_ON_ERROR is set.

File: src/e2e-test/java/io/cdap/plugin/utils/enums/NotifyOptions.java [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+/**
+ * NotifyOptions enum.
+ */
+
+public enum NotifyOptions {
+  ENABLED("Enabled"),
+  DISABLED("Disabled"),
+  WHERE("Where"),
+  ALL("All"),
+  REFERENCED("Referenced"),
```
Comment (rmstar) [inline src/e2e-test/java/io/cdap/plugin/utils/enums/NotifyOptions.java:28]: are these options used in tests - WHERE, ALL, REFERENCED?

**Pull request closed without merging**